python/src/spark_rapids_ml/umap.py

```diff
@@ -315,6 +333,7 @@ def __init__(self, sample_fraction: float = 1.0, **kwargs: Any) -> None:
         self.set_params(**kwargs)
         self.sample_fraction = sample_fraction
         self.maxRecordsPerBatch = 10000
```

Comment:
Is there any reason to hardcode the maxRecordsPerBatch to 10000?

Reply:
Just a placeholder to avoid type-check issues, since it gets set again later in _fit(). 10,000 is the default from pyspark.sql.execution.arrow.